### PR TITLE
Fixed #28727 -- Fixed Cast crash on SQlite when casting a Python date/datetime to Date/DateTimeField.

### PR DESCRIPTION
**Pull request by felixxm**
https://code.djangoproject.com/ticket/28727

### CODE REVIEW

File: tests/db_functions/test_cast.py [hunk 1]
```diff
@@ -42,6 +44,15 @@ def test_cast_to_integer(self):
                 numbers = Author.objects.annotate(cast_int=Cast('alias', field_class()))
                 self.assertEqual(numbers.get().cast_int, 1)
 
+    def test_cast_from_python_to_date(self):
+        for field_class, value in (
+            (models.DateField, datetime.date.today()),
+            (models.DateTimeField, datetime.datetime.now().replace(microsecond=0)),
```
Comment (sir-sigurd) [inline tests/db_functions/test_cast.py]: Why `replace(microsecond=0)` is needed?

Comment (felixxm) [reply tests/db_functions/test_cast.py]: Microseconds disappear on Oracle, it's a separate issue. I will try to investigate and solve it at the weekend.

Comment (felixxm) [reply tests/db_functions/test_cast.py]: @sir-sigurd I added explicit workaround on Oracle that will be removed after [#28934](https://code.djangoproject.com/ticket/28934) fix. IMO this PR should be merged in advance.

**Comment by felixxm** [comment]
buildbot, test on oracle.

File: tests/db_functions/test_cast.py [hunk 2]
```diff
@@ -42,6 +44,19 @@ def test_cast_to_integer(self):
...
 
+    def test_cast_from_python_to_date(self):
+        today = datetime.date.today()
+        dates = Author.objects.annotate(cast_date=Cast(today, models.DateField()))
+        self.assertEqual(dates.get().cast_date, today)
+
+    def test_cast_from_python_to_datetime(self):
+        now = datetime.datetime.now()
+        if connection.vendor == 'oracle':
+            # Workaround until #28934 will be fixed.
```
Comment (timgraham) [inline tests/db_functions/test_cast.py]: will be -> is

**Comment by felixxm** [comment]
@timgraham Thanks for review.

**Pull request merged**